lib/providers.dart

```diff
@@ -28,3 +28,4 @@ final customDropDownProvider =
 
 final primaryViewProvider =
     Provider.autoDispose((ref) => ref.watch(primaryViewController.notifier));
+final navController = StateProvider.autoDispose((ref) => 0);
```

Comment:
LGTM. But do we need `tabControllerProvider` anymore? I believe it's not required any more. Please look into it, and remove line 23 if it's not required any more.

Reply:
Oh my bad, forgot to remove.